crates/rpc/src/utils.rs

```diff
@@ -0,0 +1,79 @@
+use pathfinder_storage::StarknetBlocksBlockId;
+
+/// Returns true if the given [block ID](StarknetBlocksBlockId) is part of the canonical block chain.
+pub fn block_exists(
+    tx: &rusqlite::Transaction<'_>,
+    block_id: StarknetBlocksBlockId,
+) -> anyhow::Result<bool> {
+    use anyhow::Context;
+    use rusqlite::params;
+
+    match block_id {
+        StarknetBlocksBlockId::Number(number) => tx.query_row(
+            "SELECT EXISTS(SELECT 1 FROM canonical_blocks WHERE number = ?)",
+            params![number],
+            |row| row.get(0),
+        ),
+        StarknetBlocksBlockId::Hash(hash) => tx.query_row(
+            "SELECT EXISTS(SELECT 1 FROM canonical_blocks WHERE hash = ?)",
+            params![hash],
+            |row| row.get(0),
+        ),
+        StarknetBlocksBlockId::Latest => return Ok(true),
```

Comment:
nit: in theory false when db empty

Reply:
We can:
1. ignore this entirely possibility (my favorite option)
2. check the db :( -- will likely be the most called option
3. try and use rpc context to check this instead via syncing or something